### PR TITLE
Working around the doxygen-related build problems

### LINKED ISSUE #3: Autotools is not processing @INC_AMINCLUDE@ on some systems
On my Ubuntu 16.04 x86_64 and P8 ppc64le systems the autotools (autoreconf / automake is not processing the @INC_AMINCLUDE@ embedded in pveclib/Makefile.am.

This line gets passed intact to the final generated Makefile and result in a make error.

This may be related if the system has the complete list of the dependencies for doxygen installed.

For Ubuntu this seems to require doxygen, doxygen-latex, and graphviz. But might be a different list or have different pacakge names on other distro or versions.

### PR DESCRIPTION
**Pull request by munroesj52**
Working around the doxygen replated build problems.

Basically the substitution @INC_AMINCLUDE@ is not being propagated to
Makefile.in and this results in a syntax error on the final Makefile file.
Also found that need to configure with --disable-doxygen-doc or install
doxygen, doxygen-latex, and graphviz.

Fixes #3

### CODE REVIEW

**Comment by ThinkOpenly** [comment]
@munroesj52 , could you add a signed-off-by in your commit message, and maybe change the title of the commit (currently "../COMMIT_EDITMSG-20180126") to something meaningful?  :-)  Otherwise, looks fine to me.

**Comment by munroesj52** [comment]
On Mon, 2018-01-29 at 18:06 +0000, Paul Clarke wrote:
> @munroesj52 , could you add a signed-off-by in your commit message,
> and maybe change the title of the commit (currently
> "../COMMIT_EDITMSG-20180126") to something meaningful? :-) Otherwise,
> looks fine to me.
> —

I did (see attached file). Not sure why you are not seeing it:

What editor are you using for global default?

I am trying to use gedit (I am really not very good with vi). But when
I 'git commit -s', gedit pops with # comments asking to be edited and
the signed-off-by.

But the git commit command line return immediately (does not wait for
my to save the edit session or exit gedit!!!) and reports that the
commit messages is missing/empty :-/

So then I try to save the commit message and use git commit -m with the
name of the file (which does not seem to do anything useful).

What am I doing wrong?


Working around the doxygen replated build problems.

Basically the substitution @INC_AMINCLUDE@ is not being propagated to
Makefile.in and this results in a syntax error on the final Makefile file.
Also found that need to configure with --disable-doxygen-doc or install
doxygen, doxygen-latex, and graphviz.


	* Makefile.am: Drop @INC_AMINCLUDE@. Include aminclude.am from
	--srcdir.
	* aminclude.am: new file.
	* acinclude.m4: new file.

	* Makefile.in: automake generated.
	* src/Makefile.in: automake generated.
	* aclocal.m4: automake generated.
	* configure: autoreconf generated.


	Signed-off-by: Steven Munroe  <munroesj52@gmail.com>

**Comment by ThinkOpenly** [comment]
On 01/29/2018 07:34 PM, Steven Munroe wrote:
> On Mon, 2018-01-29 at 18:06 +0000, Paul Clarke wrote:
>> @munroesj52 , could you add a signed-off-by in your commit message,
>> and maybe change the title of the commit (currently
>> "../COMMIT_EDITMSG-20180126") to something meaningful? :-) Otherwise,
>> looks fine to me.
>> —
> 
> I did (see attached file). Not sure why you are not seeing it:
> 
> What editor are you using for global default?

Command line git?  I use "vi".

> I am trying to use gedit (I am really not very good with vi). But when
> I 'git commit -s', gedit pops with # comments asking to be edited and
> the signed-off-by.
> 
> But the git commit command line return immediately (does not wait for
> my to save the edit session or exit gedit!!!) and reports that the
> commit messages is missing/empty :-/
> 
> So then I try to save the commit message and use git commit -m with the
> name of the file (which does not seem to do anything useful).
> 
> What am I doing wrong?

I generally compose the commit message in a file before invoking "git commit", then use "git commit -F commit.txt".

Note that you modify the commit message with "git commit --amend", but you'll have to get the editor part working.  You'd then force the push like "git push --force origin HEAD".

**Comment by munroesj52** [comment]
On Tue, 2018-01-30 at 16:25 +0000, Paul Clarke wrote:
> On 01/29/2018 07:34 PM, Steven Munroe wrote:
> > On Mon, 2018-01-29 at 18:06 +0000, Paul Clarke wrote:
> >> @munroesj52 , could you add a signed-off-by in your commit
> message,
> >> and maybe change the title of the commit (currently
> >> "../COMMIT_EDITMSG-20180126") to something meaningful? :-)
> Otherwise,
> >> looks fine to me.
> >> —
> > 
> > I did (see attached file). Not sure why you are not seeing it:
> > 
> > What editor are you using for global default?
> 
> Command line git? I use "vi".
> 
> > I am trying to use gedit (I am really not very good with vi). But
> when
> > I 'git commit -s', gedit pops with # comments asking to be edited
> and
> > the signed-off-by.
> > 
> > But the git commit command line return immediately (does not wait
> for
> > my to save the edit session or exit gedit!!!) and reports that the
> > commit messages is missing/empty :-/
> > 
> > So then I try to save the commit message and use git commit -m with
> the
> > name of the file (which does not seem to do anything useful).
> > 
> > What am I doing wrong?
> 
> I generally compose the commit message in a file before invoking "git
> commit", then use "git commit -F commit.txt".
> 
> Note that you modify the commit message with "git commit --amend",
> but you'll have to get the editor part working. You'd then force the
> push like "git push --force origin HEAD".

Ok I did the steps above. Please let me know if it improves what you
see.

I think the problem with gedit that it insists on combining multiple
sessions. So if I already have one or more files open with gedit the
next gedit is treated as gedit &. Which confuses the git command line
tools.

**Comment by munroesj52** [comment]
On Tue, 2018-01-30 at 23:48 +0000, Paul Clarke wrote:
> @ThinkOpenly approved this pull request.
> I updated the title of the commit. I can commit it, or you may if you
> prefer.
> —

Go ahead I am going top be out of pocket most of tomorrow.

I created an issue (#2) that you can reference and close.

**Comment by ThinkOpenly** [comment]
On 01/30/2018 08:38 PM, Steven Munroe wrote:
> On Tue, 2018-01-30 at 23:48 +0000, Paul Clarke wrote:
>> @ThinkOpenly approved this pull request.
>> I updated the title of the commit. I can commit it, or you may if you
>> prefer.


> Go ahead I am going top be out of pocket most of tomorrow.

Done.  Updating the title was apparently not sufficient to change the commit title, it seems.  I'm not sure your final push ever arrived.  Anyway, it's not life-threatening.  Moving on...

> I created an issue (#2) that you can reference and close.

It was actually issue #3, which I added "Fixes #3" to your commit description, and that issue is now closed.
